### PR TITLE
feat(rtc_interface): add finish distance

### PR DESCRIPTION
**Pull request by rej55**
Signed-off-by: Fumiya Watanabe <rej55.g@gmail.com>

## Description

I add the finish distance for RTC status.
In the behavior_velocity_planner, start distance is equal to finish distance because the stop point included in the path is only one.

## Related links

This PR needs to change the msg file like following:
https://github.com/tier4/tier4_autoware_msgs/pull/50

The change for API is following:
https://github.com/tier4/tier4_ad_api_adaptor/pull/64

## Tests performed

Tested in planning_simulator.

- Lane change
Screenshot from 2022-08-29 18-21-27

```
start_distance: -0.017488522455096245
finish_distance: 12.462906837463379
```

- Avoidance

Screenshot from 2022-08-29 18-26-45

```
start_distance: 1.1015424728393555
finish_distance: 41.60154342651367
```

- Pull Over

Screenshot from 2022-08-29 18-32-44

```
start_distance: 11.645940780639648
finish_distance: 26.734411239624023
```

- Pull Out

Screenshot from 2022-08-29 18-33-45

```
start_distance: 4.999998092651367
finish_distance: 16.58200454711914
```

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/behavior_path_planner/src/scene_module/pull_over/pull_over_module.cpp [hunk 1]
```diff
@@ -626,22 +626,25 @@ BehaviorModuleOutput PullOverModule::planWaitingApproval()
...
   waitApproval();
 
   return out;
 }
 
-double PullOverModule::calcDistanceToPathChange() const
+std::pair<double, double> PullOverModule::calcDistanceToPathChange() const
 {
   const Pose parking_start_pose = getParkingStartPose();
+  const Pose parking_end_pose = getRefinedGoal();
```
Comment (kosuke55) [inline planning/behavior_path_planner/src/scene_module/pull_over/pull_over_module.cpp:639]: please use `modified_goal_pose_`

**Comment by kosuke55** [comment]
@rej55 
I think pull_out (shift_pull_out) start_distance should be almost 0.
> start_distance: 4.999998092651367
finish_distance: 16.58200454711914

**Comment by kosuke55** [comment]
>I think pull_out (shift_pull_out) start_distance should be almost 0.

My printing result. It may be ok. just not updating Description?
```
[component_container_mt-47] start_distance : 3.90419e-11
[component_container_mt-47] finish_distance : 20.2096
```

**Comment by rej55** [comment]
I didn't update description. Sorry.

File: planning/behavior_path_planner/src/scene_module/pull_over/pull_over_module.cpp (hunk 1 above)
Comment (rej55) [reply planning/behavior_path_planner/src/scene_module/pull_over/pull_over_module.cpp]: fixed https://github.com/autowarefoundation/autoware.universe/pull/1685/commits/4684c364c272cca7c48bad35098ce690abf9a1d1

**Review by kosuke55 — APPROVED** [review]
LGTM for pull_over and pull_out

**Pull request merged**